Assets/Plugins/Source/Core/Utility/FileUtility.cs

```diff
@@ -33,26 +33,80 @@ namespace PlayEveryWare.EpicOnlineServices.Utility
     public static class FileUtility
     {
         /// <summary>
-        /// Generates a unique and new temporary directory inside the Temporary Cache Path as determined by Unity,
-        /// and returns the fully-qualified path to the newly created directory.
+        /// Generates a unique and new temporary directory inside the Temporary
+        /// Cache Path as determined by Unity, and returns the fully-qualified
+        /// path to the newly created directory. The directory name will be
+        /// "Temp-" appended with a GUID.
         /// </summary>
-        /// <returns>Fully-qualified file path to the newly generated directory.</returns>
-        public static string GenerateTempDirectory()
+        /// <param name="path">
+        /// Fully-qualified path to the newly generated directory if one was
+        /// created, otherwise path is set to default System.String value.
+        /// </param>
+        /// <returns>
+        /// True if the temporary directory was created, false otherwise.
+        /// </returns>
+        public static bool TryGetTempDirectory(out string path)
         {
+            path = default;
+            
+            // Nested local function to reduce repetitive code.
+            string GenerateTempPath()
+            {
+                return Path.Combine(
+                    Application.temporaryCachePath,
+                    $"Temp-{Guid.NewGuid()}");
+            }
+
             // Generate a temporary directory path.
-            string tempDirectory = Path.Combine(Application.temporaryCachePath, $"/Output-{Guid.NewGuid()}/");
+            string tempPath = GenerateTempPath();
 
-            // If (by some crazy miracle) the directory path already exists, keep generating until there is a new one.
-            while (Directory.Exists(tempDirectory))
+            // If (by some crazy miracle) the directory path already exists,
+            // try once more.
+            if (Directory.Exists(tempPath))
             {
-                tempDirectory = Path.Combine(Application.temporaryCachePath, $"/Output-{Guid.NewGuid()}/");
+                Debug.LogWarning(
+                    $"The temporary directory created collided with " +
+                    $"an existing temporary directory of the same name. This " +
+                    $"is very unlikely.");
```

Comment:
not a critique: very, very, very unlikely. Something very strange and wrong must happen for a guid collision, ahah.